packages/api/cms-api/src/auth/guards/comet.guard.ts

```diff
@@ -34,29 +36,36 @@ export class CometAuthGuard implements CanActivate {
             return true;
         }
 
-        let user = await this.getAuthenticatedUser(request);
-        if (!user) return false;
+        const result = await this.getAuthenticatedUser(request);
+        if (!result) return false;
+        if ("authenticationError" in result) throw new UnauthorizedException(result.authenticationError);
```

Comment:
Is it okay to throw an HTTP Error here? Shouldn't we return false instead? The guard will be used for GraphQL as well.